src/Perl6/Pod.nqp

```diff
@@ -116,32 +106,281 @@ class Perl6::Pod {
         ).compile_time_value
     }
 
+    sub convert-array(@raw) {
+        # input is an array of strings to be converted to a list or hash
+        my @arr := nqp::list();
+        for @raw -> $s {
+            #my $s := $S;
+            say("DEBUG: element to be converted: |$s|") if $debugp;
+
+            # Convert the strings of numbers or True|False to int,
+            # bigint, num, or bool types.
+            if $s ~~ /^ <[+-]>? \d+ $/ {
+                my $n := nqp::chars($s);
+                # delete 1 char for leading +-
+                if $s ~~ / <[+-]> / { --$n }
+
+                # convert the int to a bigint if required
+                #   max val int (int32): +2_147_483_647 <= 10 digits
+                #   max val int (int64): +9_223_372_036_854_775_807 <= 19 digits
+                # we convert to bigint after 9 digits
+                # we convert to string after 18 digits
+                if $n > 18 {
```

Comment:
There has to be a better way. Also, I'm assuming this is not going to work on 32-bit systems?

Reply:
What do you mean by “too big for bigint”? I thought the point of bigints is to store unlimitedly huge ints :)

Reply:
well, my limited testing showed a limit of a 64-bit integer...

Reply:
I guess the misunderstanding is that int64 is a bigint? It's not. A bigint is just an `Int`, and I think that the logic that determines if it it fits into a native int or not is unnecessary. Let's keep the code as simple as possible. Or, if not, why can't it be just an Int?

Reply:
wait, forget what i just said. after looking at the opcodes again i think i see how to do the conversion. it has to get specifically get “boxed” as an Int...

Reply:
Bigints are now handled properly with the latest commit.